### PR TITLE
Prevent notifications DELETE

### PR DESCRIPTION
**Pull request by jayvdb**
HTTP method DELETE causes a TypeError, as the delete operation
must be done via POST.

Fixes https://github.com/django-oscar/django-oscar/issues/3431

### CODE REVIEW

**Comment by jayvdb** [comment]
Note there is also a smoke test error wrt GET described at https://github.com/django-oscar/django-oscar/issues/3428#issuecomment-650498588

Happy to include the fix here, with a bit of guidance on what type of fix would be acceptable.

**Comment by jayvdb** [comment]
CI fails with `isort: error: unrecognized arguments: --recursive`

**Comment by solarissmoke** [comment]
> CI fails with `isort: error: unrecognized arguments: --recursive`

Fixed in 3b7a1bc (you'll need to rebase this PR onto master).

> Note there is also a smoke test error wrt GET described

That is coming from the parent `RedirectMixin`, which actually isn't being used here at all (we never define a redirect URL), so it seems to have been a mistake to subclass it. We can just remove that mixin. The implication of that is that only `POST` is allowed on this view.

File: src/oscar/apps/communication/notifications/views.py [hunk 1]
```diff
@@ -69,8 +69,9 @@ def get_queryset(self):
             recipient=self.request.user)
 
 
-class UpdateView(BulkEditMixin, generic.RedirectView):
+class UpdateView(BulkEditMixin, generic.View):
     model = Notification
+    http_method_names = ['post']
     actions = ('archive', 'delete')
     checkbox_object_name = 'notification'
     permanent = False
```
Comment (solarissmoke) [inline src/oscar/apps/communication/notifications/views.py:77]: Please remove this line (`permanent = False`) - it's also a hangover from the redirect view.

**Pull request merged**